packages/e2e.sandbox.views.todo-widget/src/app.tsx

```diff
@@ -0,0 +1,86 @@
+import { ExclamationTriangleIcon } from "@radix-ui/react-icons";
+import {
+  Box,
+  Button,
+  Callout,
+  Container,
+  Flex,
+  Heading,
+  Skeleton,
+  Table,
+} from "@radix-ui/themes";
+import React, { useCallback, useEffect } from "react";
+import { useFoundryContext } from "./context.js";
+
+export const App: React.FC = () => {
+  const { parameterValues, hostEventTarget, emitEvent } = useFoundryContext();
+  const { headerText, showWarning } = parameterValues;
+
+  useEffect(() => {
+    hostEventTarget.addEventListener("host.update-parameters", (event) => {
+      console.log("Received event:", event);
+    });
+  }, []);
+
+  const handleAddRow = useCallback(() => {
+    emitEvent("view.emit-event", {
+      eventId: "todo",
```

Comment:
i realized i forgot to handle the types for events, but i'll do that in a follow up PR instead